Neo4j.Driver/Neo4j.Driver/Internal/Connector/SocketConnection.cs

```diff
@@ -38,17 +39,27 @@ internal class SocketConnection : IConnection
         private readonly Queue<IRequestMessage> _messages = new Queue<IRequestMessage>();
         internal IReadOnlyList<IRequestMessage> Messages => _messages.ToList();
 
-        private readonly ILogger _logger;
+        private readonly IDriverLogger _logger;
+
+        private readonly string _id;
 
         public SocketConnection(Uri uri, ConnectionSettings connectionSettings, BufferSettings bufferSettings,
-            IConnectionListener metricsListener = null, ILogger logger = null)
-            : this(new SocketClient(uri, connectionSettings.SocketSettings, bufferSettings, metricsListener, logger),
-                connectionSettings.AuthToken, connectionSettings.UserAgent, logger, new ServerInfo(uri))
+            IConnectionListener metricsListener = null, IDriverLogger logger = null)
         {
+            _id = $"Conn-{UniqueIdGenerator.GetId()}";
```

Comment:
And also here, and even we can make use of `connection_id` assigned by the server on Bolt V3?